app/domain/authentication/authn_oidc/v2/strategy.rb

```diff
@@ -15,17 +15,27 @@ def initialize(
 
         # Don't love this name...
         def callback(args)
```

Comment:
I think part of the reason why the `AuthenticationHandler` class was created was to see if it could be adopted as the new standard for all our authn schemes (see commit 1c458bbb7d32ffba8e29f87ba27c4e33e910019b in [this draft PR](https://github.com/cyberark/conjur/pull/2663) from @jvanderhoof).

Given that this `Strategy` class could become a catch-all for what could be described as 3 distinct strategies, maybe we should create a class for each:
```rb
module Strategies
  class Code
    ...
  end

  class RefreshToken
    ...
  end

  class IdToken
    ...
  end

  class Common
    ...
  end
end
```